creator-node/src/utils.js

```diff
@@ -16,12 +16,17 @@ class Utils {
 
 async function getFileUUIDForImageCID (req, imageCID) {
   const ipfs = req.app.get('ipfsAPI')
+  const ipfsLatest = req.app.get('ipfsLatestAPI')
 
   if (imageCID) { // assumes imageCIDs are optional params
     // Ensure CID points to a dir, not file
     let cidIsFile = false
     try {
-      await ipfs.cat(imageCID, { length: 1 })
+      const chunks = []
+      for await (const chunk of ipfsLatest.cat(imageCID, { length: 1, timeout: 1000 })) {
```

Comment:
does ipfs.cat return >1 entry if the multihash links to others? lets comment explaining why we need this and not just a single cat op

Reply:
@hareeshnagaraj this is because of the way the new cat api behaves. this is returning this thing called async iterator, not a promise, so you can't just await ipfs.cat. the api usage is now https://github.com/ipfs/interface-js-ipfs-core/blob/master/SPEC/FILES.md#cat

Reply:
got it